Assets/MRTK/SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -75,7 +75,13 @@ public float MaxRaycastDistance
         public float SurfaceNormalOffset
         {
             get => surfaceNormalOffset;
-            set => surfaceNormalOffset = value;
+            set
+            {
+                // If a user were to configure Tap to Place via script and they try to set the SurfaceNormalOffset while UseDefaultSurfaceNormalOffset is true, display the following error:
+                Debug.Assert(!UseDefaultSurfaceNormalOffset, $"The new value for SurfaceNormalOffset on the Tap to Place object will not be applied because UseDefaultSurfaceNormalOffset is true, set UseDefaultSurfaceNormalOffset to false.");
+
```

Comment:
What are the thoughts on adding this warning?

1. UseDefaultSurfaceNormalOffset is true by default and I can see someone adding the tap to place component via script, then try to modify the SurfaceNormalOffset and then wonder why their value is not getting applied.  This is not an issue via inspector because the SurfaceNormalOffset property is hidden until UseDefaultSurfaceNormalOffset is set to false.

2. Although, I also see a case where someone already knows this information and is just setting initial properties to eventually toggle between the defaultSurfaceNormalOffset and their set SurfaceNormalOffset.  Then this message in the console would be unnecessary.

Reply:
Yeah I don't think that this is going to be too problematic (i.e. I think 2 is what's gonna happen) but at the same time, I don't think that this assert will hurt (and could help in case 1), so it's fine to leave in IMO.